File: Python/1207. Unique Number of Occurrences.py
from collections import Counter 
class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        occur = Counter(arr)
        counts = set()
        for _, v in occur.items():
            if v in counts:
                return False
            counts.add(v)
        return True
        
        

File: 2024/1207. Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> counts;
        for (int n : arr) {
            counts[n]++;
        }
        unordered_set<int> freq;
        for (auto& [_, f] : counts) {
            if (freq.find(f) != freq.end()) {
                return false;
            }
            freq.insert(f);
        }
        return true;
    }
};